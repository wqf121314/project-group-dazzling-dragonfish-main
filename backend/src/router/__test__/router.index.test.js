const server = require('../../server').callback();
const request = require('supertest')

describe('Checking the main interface', () => {
    it('Get not exit request, interface: / ', async () => {
        const response = await request(server).get('/');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();
    })
    it('Post not exit request, interface: / ', async () => {
        const response = await request(server).post('/');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('Get not exit request, interface: /index  ', async () => {
        const response = await request(server).get('/index');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('Post not exit request, interface: /index  ', async () => {
        const response = await request(server).get('/index');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
});

describe('GET /api', () => {
    it('check error interface: /api ', async () => {
        const response = await request(server).get('/api');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();
    })
    it('checking for api/health', async () => {
        const response = await request(server).get('/api/health');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(0);
        expect(response.body.msg).toEqual('');
        expect(response.body.data).toEqual('Backend Server is running');
        expect(response.body).toMatchSnapshot();
    })
});

describe('GET /api/v1', () => {
    it('check error interface: /api/v1 ', async () => {
        const response = await request(server).get('/api/v1');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('check error interface: /api/v1/index ', async () => {
        const response = await request(server).get('/api/v1/index');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('checking for api/v1/health', async () => {
        const response = await request(server).get('/api/v1/health');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(0);
        expect(response.body.msg).toEqual('');
        expect(response.body.data).toEqual('Backend Server V1 is running');
        expect(response.body).toMatchSnapshot();
    })
});

describe('GET /api/v2', () => {
    it('check error interface: /api/v2 ', async () => {
        const response = await request(server).get('/api/v2');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('check error interface: /api/v2/index ', async () => {
        const response = await request(server).get('/api/v2/index');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(1);
        expect(response.body.msg).toEqual('URL is not exist!!');
        expect(response.body).toMatchSnapshot();

    })
    it('checking for api/v2/health', async () => {
        const response = await request(server).get('/api/v2/health');
        expect(response).toBeDefined();
        expect(response.status).toEqual(200);
        expect(response.type).toEqual('application/json');
        expect(response.body.code).toEqual(0);
        expect(response.body.msg).toEqual('');
        expect(response.body.data).toEqual('Backend Server V2 is running');
        expect(response.body).toMatchSnapshot();
    })
});
