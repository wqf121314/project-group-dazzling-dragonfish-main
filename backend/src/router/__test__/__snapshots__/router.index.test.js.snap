// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking the main interface Get not exit request, interface: /  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`Checking the main interface Get not exit request, interface: /index   1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`Checking the main interface Post not exit request, interface: /  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`Checking the main interface Post not exit request, interface: /index   1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api check error interface: /api  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api checking for api/health 1`] = `
Object {
  "code": 0,
  "data": "Backend Server is running",
  "msg": "",
}
`;

exports[`GET /api/v1 check error interface: /api/v1  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api/v1 check error interface: /api/v1/index  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api/v1 checking for api/v1/health 1`] = `
Object {
  "code": 0,
  "data": "Backend Server V1 is running",
  "msg": "",
}
`;

exports[`GET /api/v2 check error interface: /api/v2  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api/v2 check error interface: /api/v2/index  1`] = `
Object {
  "code": 1,
  "msg": "URL is not exist!!",
}
`;

exports[`GET /api/v2 checking for api/v2/health 1`] = `
Object {
  "code": 0,
  "data": "Backend Server V2 is running",
  "msg": "",
}
`;
